tests/serverpod_test_server/test_integration/auth_core/auth_header_test.dart

```diff
@@ -73,5 +73,61 @@ void main() async {
       var receivedToken = await tokenInspectionCompleter.future;
       expect(receivedToken, key);
     });
+
+    test(
+        'then old clients passing keys in body should still work with new server code',
+        () async {
+      var key = 'username-4713:password-4713';
+
+      // Intentionally clearing the authKeyManager so that auth is not passed in header.
+      await authKeyManager.remove();
+
+      // Verify that by calling it this way, the key is not passed in the header.
+      // This validates this unit test's approach.
+      tokenInspectionCompleter = Completer();
+      var reflectedHeader =
+          await client.reflection.caller.callServerEndpoint<List<String>?>(
+        'reflection',
+        'reflectHttpHeader',
+        {
+          'headerName': 'authorization',
+          'auth': key,
+        },
+      );
```

Comment:
Suggestion: Use a pure HTTP client to make this request.

Given that changes are made to `callServerEndpoint`, the format of this call might be mutated in such a way that it no longer reflects the "old" approach.
A safer approach would be to bypass our client generated code and simply pass in the data structure we expect to be correctly parsed by the server.

Reply:
Yes! See auth_core_test.dart